src/buffer/out/textBuffer.cpp

```diff
@@ -1260,6 +1260,94 @@ bool TextBuffer::MoveToPreviousWord(COORD& pos, std::wstring_view wordDelimiters
     return true;
 }
 
+// Method Description:
+// - Update pos to be the beginning of the current glyph/character. This is used for accessibility
+// Arguments:
+// - pos - a COORD on the word you are currently on
+// Return Value:
+// - pos - The COORD for the first cell of the current glyph (inclusive)
+const til::point TextBuffer::GetGlyphStart(const til::point pos) const
+{
+    COORD resultPos = pos;
+
+    const auto bufferSize = GetSize();
+    if (resultPos == bufferSize.EndExclusive() || GetCellDataAt(resultPos)->DbcsAttr().IsTrailing())
+    {
+        bufferSize.DecrementInBounds(resultPos, true);
+    }
+
+    return resultPos;
+}
+
+// Method Description:
+// - Update pos to be the end of the current glyph/character. This is used for accessibility
+// Arguments:
+// - pos - a COORD on the word you are currently on
+// Return Value:
+// - pos - The COORD for the last cell of the current glyph (exclusive)
+const til::point TextBuffer::GetGlyphEnd(const til::point pos) const
+{
+    COORD resultPos = pos;
+
+    const auto bufferSize = GetSize();
+    if (resultPos == bufferSize.EndExclusive() || GetCellDataAt(resultPos)->DbcsAttr().IsLeading())

```

Comment:
won't incrementing past this EndExclusive explode?

Reply:
Nope. We actually need this check like this. If we're at EndExclusive, do NOT check the non-existent cell data and  "increment". Incrementing here is safe because we just return false in "IncrementInBounds". So we're good.

Reply:
I guess it could be rewritten as `resultPos != bufferSize.EndExclusive() && GetCellDataAt(...` for clarity.